docs/index.rst

```diff
@@ -158,8 +158,9 @@ There are some additional tools, mostly of use for maintainers, in the
 
 {%else%}
 
-To read the developer documentation, you will need to go to the
-`latest developer version of the documentation <https://docs.astropy.org/en/latest/#developer-documentation>`__.
+To read the developer documentation, you will need to go to the :ref:`latest
+developer version of the documentation
+<astropy-dev:developer-documentation>`.
```

Comment:
This didn't render as "stable": `docs/index.rst:133: WARNING: undefined label: developer-documentation`

Reply:
I'm not sure what you mean here.  In this case it should be using intersphinx to resolve to latest/, so long as that label actually exists in latest/

Reply:
But it failed. Do you know why?

Reply:
Ah, I think I figured it out. The URL points to heading title, but now we need to point to Sphinx ref tag: https://github.com/astropy/astropy/pull/11622/files#r623313262